spec/unit/policyfile_includes_spec.rb

```diff
@@ -0,0 +1,512 @@
+        end
+      end
+
+      context "and the including policy specifies a source that is not equivalent to the included policy" do
+        let(:run_list) { [] }
+        let(:default_source) { instance_double("ChefDK::Policyfile::NullCookbookSource") }
+
+        before do
+          allow(default_source).to receive(:preferred_cookbooks).and_return(["cookbookC"])
+          allow(default_source).to receive(:source_options_for).with("cookbookC", "2.0.0").and_return({ "foo" => "bar" })
+          allow(default_source).to receive(:null?).and_return(false)
+          allow(default_source).to receive(:universe_graph).and_return(external_cookbook_universe)
+          allow(default_source).to receive(:desc).and_return("source double")
+        end
+
+        it "it raises an error" do
+          expect { policyfile_lock.to_lock }.to raise_error(ChefDK::IncludePolicyCookbookSourceConflict)
+        end
+      end
+
+      context "and the including policy's dependencies can be solved with the included policy's locks" do
+        let(:run_list) { ["local_easy::default"] }
+
+        it "solves the dependencies added by the top-level policyfile and emits them in the lockfile" do
+          expect(policyfile_lock.to_lock["solution_dependencies"]["dependencies"]).to(
+            have_key("cookbookC (2.0.0)"))
+        end
+
+      end
+
+      context "and the including policy's dependencies cannot be solved with the included policy's locks" do
+        let(:run_list) { ["local::default"] }
+
+        it "raises an error describing the conflict" do
+          expect { policyfile_lock.to_lock }.to raise_error(Solve::Errors::NoSolutionError)
+        end
+
+        it "includes the name and location of the included policy in the error message" do
+          pending("Not implemented")
+          expect { policyfile_lock.to_lock }.to raise_error(Solve::Errors::NoSolutionError) do |e|
+            expect(e.to_s).to match(/#{included_policy_lock_name}/)
+          end
+
+        end
+
+        it "includes the source of the conflicting dependency constraint from the including policy" do
+          expect { policyfile_lock.to_lock }.to raise_error(Solve::Errors::NoSolutionError) do |e|
+            expect(e.to_s).to match(/`cookbookC \(= 2.0.0\)`/) # This one comes from the included policy
+            expect(e.to_s).to match(/`cookbookC \(= 1.0.0\)` required by `local-1.0.0`/) # This one comes from the included policy
+          end
+        end
+
+        it "should not have an open constraint on cookbookC" do
+          pending("I think this is a bug")
+          expect { policyfile_lock.to_lock }.to raise_error(Solve::Errors::NoSolutionError) do |e|
+            expect(e.to_s).not_to match(/`cookbookC \(>= 0.0.0\)`/)
+          end
+        end
+
+      end
+
+      context "and the including policy explicitly sets the source for a cookbook which conflicts with the source in the included policy" do
```

Comment:
we adding these tests?

Reply:
I think these are fairly well covered by `spec/unit/policyfile/included_policies_cookbook_source_spec.rb`. Can remove this